File: SeongJae/week2/BOJ_9935.py
import sys
from collections import deque
sentence=sys.stdin.readline().rstrip()
bomb=sys.stdin.readline().rstrip()
stack=[]
txt=""
if len(bomb)==1:
    txt=""
    for i in sentence:
        if i!=bomb[0]:
            txt+=i
else:
    for i in sentence:
        #첫글자 이면
        if i==bomb[0]:
            stack.append(txt)
            txt=bomb[0]
        #끝이면
        elif i==bomb[-1]:
            txt+=i
            if txt==bomb:
                if len(stack)>0:
                    txt=stack[-1]
                    stack.pop()
        else:
            txt+=i
if txt!="":
    stack.append(txt)
if stack:
    for i in stack:
        print(i,end="")
else:
    print("FRULA")

File: yubin/2st/BOJ_9935.java
package test;

import java.io.BufferedReader;
import java.io.FileInputStream;
import java.io.InputStreamReader;
import java.util.Stack;

public class BOJ_9935 {
	
	public static void main(String[] args) throws Exception{
		System.setIn(new FileInputStream("res/input.txt"));
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		String str = br.readLine();
		String bomb = br.readLine();
				
		Stack<Character> stack = new Stack<>();
		
		
		for (int i = 0; i < str.length(); i++) {
			stack.push(str.charAt(i));
			
			if(stack.size() >= bomb.length()) {
				for (int j = 0; j < bomb.length(); j++) {
					if(stack.get(stack.size() - bomb.length() + j) != bomb.charAt(j)) {
						break;
					}
					if(j == bomb.length() - 1) {
						for (int j2 = 0; j2 < bomb.length(); j2++) {
							stack.pop();
						}
					}
				}
			}
		}
		
		if(stack.isEmpty()) {
			System.out.println("FRULA");
			return;
		}
		
		StringBuilder sb = new StringBuilder();
		for (int i = 0; i < stack.size(); i++) {
			sb.append(stack.get(i));
		}
		
		System.out.println(sb);

		
	}
}
